yarn-project/pxe/src/database/kv_pxe_database.ts

```diff
@@ -165,27 +165,47 @@ export class KVPxeDatabase implements PxeDatabase {
     }
   }
 
+  *#getAllNonNullifiedNoteIndices(): IterableIterator<number> {
+    for (const [index, _] of this.#notes.entries()) {
+      if (this.#nullifiedNotes.has(index)) {
+        continue;
+      }
+
+      yield index;
+    }
+  }
+
   async getNotes(filter: NoteFilter): Promise<NoteDao[]> {
     const publicKey: PublicKey | undefined = filter.owner
       ? (await this.getCompleteAddress(filter.owner))?.publicKey
       : undefined;
 
-    const initialNoteIds = publicKey
-      ? this.#notesByOwner.getValues(publicKey.toString())
-      : filter.txHash
-      ? this.#notesByTxHash.getValues(filter.txHash.toString())
-      : filter.contractAddress
-      ? this.#notesByContract.getValues(filter.contractAddress.toString())
-      : filter.storageSlot
-      ? this.#notesByStorageSlot.getValues(filter.storageSlot.toString())
-      : undefined;
+    // TODO: refactor how we create the array of candidate note ids once the kv_pxe_database refactor from #3927 is
+    // merged.
+    let candidateNoteIds: Array<number> = [];
+
+    filter.status = filter.status ?? 'activeOnly';
+
+    if (filter.status == 'activeOnly' || filter.status == 'includeNullified') {
```

Comment:
What is the benefit of this if? It includes both cases non-undefined cases and if undefined you will set the value to `activeOnly` on 187.


What more options did you have for the status? Also, can we use an enum or something instead, seems very easy to miss